### PR TITLE
SDI-311 Elastic search for establishment search

### PR DESCRIPTION
**Pull request by andymarke**
1. Adds integration to `prisoner-offender-search` for the main DPS (establishment) search
2. Existing `prison-api` search is used by default
3. `prisoner-offender-search` can be used by adding `feature=new` to any /prisoner-search url
4. Old behaviour is replicated

### CODE REVIEW

File: docker-compose-test.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: '3.8'
 services:
   digital-prison-services-wiremock:
-    image: rodolpheche/wiremock
+    image: wiremock/wiremock
     container_name: digital-prison-services-test-wiremock
```
Comment (andymarke) [inline docker-compose-test.yaml:5]: official version (arm compatible)

File: backend/api/offenderSearchApi.ts [hunk 2]
```diff
@@ -13,6 +14,16 @@ export type PrisonerSearchResult = {
   cellLocation: string
   status: string
   alerts: Array<Alert>
+  category: string
+}
```
Comment (andymarke) [inline backend/api/offenderSearchApi.ts:18]: correction to exiting type, it is called `category`

File: backend/tests/prisonerSearch.test.ts [hunk 3]
```diff
@@ -455,14 +476,432 @@ describe('Prisoner search', () => {
...
-            'Page-Limit': 500,
+            'page-limit': 500,
           }),
         }),
         'MDI',
         { alerts: undefined, keywords: undefined, returnAlerts: 'true', returnCategory: 'true' }
       )
     })
   })
+  describe('index (search-api)', () => {
```
Comment (andymarke) [inline backend/tests/prisonerSearch.test.ts:487]: essentially a copy of above but obviously calling the search api instead

File: backend/tests/prisonerSearch.test.ts [hunk 4]
```diff
@@ -455,14 +476,432 @@ describe('Prisoner search', () => {
...
+      expect(prisonApi.userLocations).toHaveBeenCalledWith(res.locals)
+    })
+
+    it('should request the current users active case load prisoners if no location specified in the query', async () => {
+      await controller.index(req, res)
+
+      expect(offenderSearchApi.establishmentSearch).toHaveBeenCalledWith(expect.anything(), 'MDI', expect.anything())
+    })
+
+    it('should request with location when supplied in query', async () => {
```
Comment (andymarke) [inline backend/tests/prisonerSearch.test.ts:506]: new test

File: integration-tests/integration/search/prisonerSearch.cy.js [hunk 5]
```diff
@@ -41,235 +16,559 @@ context('Prisoner search', () => {
...
+            .then(($tableRows) => {
+              cy.get($tableRows).find('a').click()
+            })
+        })
+
+        cy.get('[data-test="recent-search-link"]').should('have.attr', 'href', searchUrl)
       })
     })
+  })
+
```
Comment (andymarke) [inline integration-tests/integration/search/prisonerSearch.cy.js:295]: essentially a copy of above but with feature switch on

File: integration-tests/integration/search/prisonerSearch.cy.js [hunk 6]
```diff
@@ -41,235 +16,559 @@ context('Prisoner search', () => {
...
-        count: 2,
-        data: [inmate1, inmate2],
+        cy.get('[data-test="prisoner-search-results-table"]').then(($table) => {
+          cy.get($table)
+            .find('tr')
+            .then(($tableRows) => {
+              cy.get($tableRows).its('length').should('eq', 3) // 2 results plus table header
+            })
+        })
       })
```
Comment (andymarke) [inline integration-tests/integration/search/prisonerSearch.cy.js:363]: new test below

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**